integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/rule/InvalidRowPercentRuleImpl.java

```diff
@@ -0,0 +1,135 @@
+package com.thinkbiganalytics.spark.dataquality.rule;
+
+/*-
+ * #%L
+ * kylo-spark-job-dataquality-app
+ * %%
+ * Copyright (C) 2017 ThinkBig Analytics
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.thinkbiganalytics.spark.dataquality.util.DataQualityConstants;
+import com.thinkbiganalytics.spark.dataquality.util.FlowAttributes;
+import com.thinkbiganalytics.spark.dataquality.util.MissingAttributeException;
+
+import org.json.simple.JSONObject;
+
+/**
+ * Data Quality Rule to ensure that: <br>
+ * Percent of Invalid rows out of all rows less than invalid threshold defined by
+ * DQ_INVALID_ALLOWED_PERCENT_ATTRIBUTE
+ * 
+ * If no threshold attribute is defined, the default value is used which is 0% i.e. no invalid rows.
+ */
+public class InvalidRowPercentRuleImpl implements DataQualityRule {
+
+    private static final Logger log = LoggerFactory.getLogger(InvalidRowPercentRuleImpl.class);
+
+    private String name;
+    private String description;
+    private boolean status;
+
+    private long feedRowCount;
+    private long invalidRowCount;
+    private long invalidThreshold;
+    private double pctInvalid;
+
+    public InvalidRowPercentRuleImpl() {
+        this.name = "INVALID_ROW_PERCENT_RULE";
+        this.description = "Percent of (Invalid rows / Total rows) <= " +
+                           DataQualityConstants.DEFAULT_INVALID_ALLOWED_PERCENT_VALUE;
+        this.status = false;
+        this.pctInvalid = Double.MAX_VALUE;
+    }
+
+    @Override
+    public boolean loadAttributes(FlowAttributes flowAttr) {
+        log.info("Loading Attributes for rule: " + name);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2629)